histogram.go

```diff
@@ -236,17 +235,19 @@ func (m *histogram) Exemplar(n int) *Exemplar {
 	return v
 }
 
-func (m *histogram) search(val float64) int {
-	if len(m.bounds) > 50 {
-		return sort.SearchFloat64s(m.bounds, val)
-	}
+// incrementBuckets increments affected buckets and returns one specific to given val.
+func (m *histogram) incrementBuckets(val float64) ( /* matching bucket */ bk *histogramBucket) {
+	// increment +Inf bucket
+	bk = &m.buckets[len(m.bounds)]
+	bk.count++
 
-	for i, b := range m.bounds {
-		if val < b {
-			return i
-		}
+	// increment matching buckets (in descending order)
+	for i := len(m.bounds) - 1; i >= 0 && val <= m.bounds[i]; i-- {
```

Comment:
that doesn't work if the value is larger than the largest boundary, i.e. between the N-1 and N (= `+Inf`) buckets